src/app/components/DynamicContentStyles.tsx

```diff
@@ -20,29 +26,44 @@ const DynamicContentStyles = React.forwardRef<HTMLElement, DynamicContentStylesP
   { children, disable, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>,
   ref
 ) => {
-  const queryParams = useSelector(query);
   const [styles, setStyles] = React.useState('');
+  const queryParams = useSelector(query);
+  const currentBook = useSelector(book);
 
   React.useEffect(() => {
     if (disable) {
       setStyles('');
       return;
     }
 
-    const cssfileUrl = queryParams['content-style'];
+    let cssfileUrl = queryParams['content-style'];
+
+    if (!cssfileUrl && currentBook && currentBook.style_href) {
+      const bookConfig = BOOKS[currentBook.id];
+      if (bookConfig && bookConfig.dynamicStyles) {
+        cssfileUrl = currentBook.style_href;
+
+        if (!isAbsoluteUrl(cssfileUrl)) {
+          const archiveUrl = `${bookConfig.archiveOverride || REACT_APP_ARCHIVE_URL}/`;
```

Comment:
Just looked at the same thing with @jivey. You can use [createBookConfigLoader.ts](https://github.com/openstax/rex-web/blob/main/src/gateways/createBookConfigLoader.ts#L44) for this and maybe some of the previous logic. `getBookVersionFromUUIDSync` returns the latest config for a given book.

Reply:
I added calls to this method and getArchiveUrl() to get the config and the default archive url.
It seems the method can return undefined if the book config is not loaded, so I added it to the useEffect array to hopefully get it to re-render properly...

Reply:
I don't see how either would return undefined - the local book config and archive URL are used until you attempt to load a new book content version, in which case the remote config is loaded and they're updated.